src/app/qgsprojectproperties.cpp

```diff
@@ -2633,34 +2641,40 @@ void QgsProjectProperties::addScaleToScaleList( QListWidgetItem *newItem )
   QListWidgetItem *duplicateItem = lstScales->findItems( newItem->text(), Qt::MatchExactly ).value( 0 );
   delete duplicateItem;
 
-  int newDenominator = newItem->text().split( ':' ).value( 1 ).toInt();
+  int newDenominator = newItem->data( Qt::UserRole ).toInt();
```

Comment:
```suggestion
  const int newDenominator = newItem->data( Qt::UserRole ).toInt();
```

Reply:
And why int when we treat them as double in the rest of the code?

Reply:
I actually asked myself the same question and decided not to change the old implementation (`toInt()`) : it's a denominator so is should definitely be an int, but  `void QgsProjectViewSettings::setMapScales( const QVector<double> &scales )`.

I don't think it makes any sense to handle doubles in the GUI (i.e. when displaying the scales or when editing/entering them).

Reply:
Ok :+1: